### PR TITLE
Allow overriding all Guzzle configuration settings

### PR DESCRIPTION
**Pull request by c960657**
This patch makes it possible to override all Guzzle configuration settings, not just user-agent and middleware.

### CODE REVIEW

File: tests/unit/Guzzle/ClientFactoryTest.php [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+			};
+		};
+
+		$clientFactory = new ClientFactory( array( 'middleware' => array( $middleware ) ) );
+
+		$client = $clientFactory->getClient();
+
+		$this->assertNull( $client->getConfig('middleware') );
+
+		$request = $this->getMockBuilder( 'Psr\Http\Message\RequestInterface' )->getMock();
```
Comment (JeroenDeDauw) [inline tests/unit/Guzzle/ClientFactoryTest.php]: You can use `RequestInterface:class`

Comment (c960657) [reply tests/unit/Guzzle/ClientFactoryTest.php]: Done.

File: tests/unit/Guzzle/ClientFactoryTest.php [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+namespace Mediawiki\Api\Test\Unit\Guzzle;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\HandlerStack;
+use Mediawiki\Api\Guzzle\ClientFactory;
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * @author Addshore
```
Comment (addshore) [inline tests/unit/Guzzle/ClientFactoryTest.php]: Am I the author of this class? ;)

File: tests/unit/Guzzle/ClientFactoryTest.php [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+ *
+ * @todo test interaction with logger
```
Comment (addshore) [inline tests/unit/Guzzle/ClientFactoryTest.php]: Again I think this comment was probably copied over in error, there isn't much interaction with the logger class in here to test!

**Review by addshore — CHANGES_REQUESTED** [review]

File: tests/unit/Guzzle/ClientFactoryTest.php (hunk 3 above)
Comment (c960657) [reply tests/unit/Guzzle/ClientFactoryTest.php]: Fixed :-)

File: tests/unit/Guzzle/ClientFactoryTest.php (hunk 2 above)
Comment (JeroenDeDauw) [reply tests/unit/Guzzle/ClientFactoryTest.php]: You are the one that will be blamed for bugs now ;)

**Pull request merged**